### PR TITLE
Implement consistency test

### PR DESCRIPTION
**Pull request by boaz-codota**
Hopefully I understood correctly what you where meaning when you said "consistency test".

Let me know if it OK.

Ref https://github.com/marceloboeira/voik/issues/8

### CODE REVIEW

**Comment by marceloboeira** [comment]
@boaz-codota yes! great, the CRC Digest should do the job. 

I've noticed the build is failing: 

```
thread 'test_commit_log_data_consistency_of_random_values' panicked at 'called `Result::unwrap()` on an `Err` value: Segment(Index(Io(Os { code: 24, kind: Other, message: "Too many open files" })))', tests/integration_test.rs:22:46
```

Usually, for integration tests we would probably need to set something like:

```
ulimit -n 10000
```

**Review by marceloboeira — APPROVED** [review]
As soon as the build is green we can merge it!

File: commit_log/.gitignore [hunk 1]
```diff
@@ -8,3 +8,6 @@ target/
 
 ## Cargo lock
 Cargo.lock
+
+# IDE
+**/.idea/
```
Comment (marceloboeira) [inline commit_log/.gitignore]: I don't mind it much, but you could move this to your global git ignore since it is personal preference.

**Comment by boazberman** [comment]
> @boaz-codota yes! great, the CRC Digest should do the job.
> 
> I've noticed the build is failing:
> 
> ```
> thread 'test_commit_log_data_consistency_of_random_values' panicked at 'called `Result::unwrap()` on an `Err` value: Segment(Index(Io(Os { code: 24, kind: Other, message: "Too many open files" })))', tests/integration_test.rs:22:46
> ```
> 
> Usually, for integration tests we would probably need to set something like:
> 
> ```
> ulimit -n 10000
> ```

I'm not sure what you by  that `ulimit` thing. Can you explain it a bit further to me?

**Comment by marceloboeira** [comment]
> > @boaz-codota yes! great, the CRC Digest should do the job.
> > I've noticed the build is failing:
> > ```
> > thread 'test_commit_log_data_consistency_of_random_values' panicked at 'called `Result::unwrap()` on an `Err` value: Segment(Index(Io(Os { code: 24, kind: Other, message: "Too many open files" })))', tests/integration_test.rs:22:46
> > ```
> > 
> > 
> > Usually, for integration tests we would probably need to set something like:
> > ```
> > ulimit -n 10000
> > ```
> 
> I'm not sure what you by that `ulimit` thing. Can you explain it a bit further to me?

the error is that there are too many open files, which is likely related to the an OS limit. In theory, we have to fix that instead of raising the limit, but for the sake of the test now just raising the limit should do the job.

ulimit does that...

**Comment by boaz-codota** [comment]
@marceloboeira fixed

**Comment by marceloboeira** [comment]
Thanks a lot!

**Pull request merged**